integrations/helm/releasesync/releasesync.go

```diff
@@ -0,0 +1,349 @@
+}
+
+// getCustomResources retrieves FluxHelmRelease resources
+//		and outputs them organised by namespace and: Chart release name or Custom Resource name
+//						map[namespace] = []ReleaseFhr
+func (rs *ReleaseChangeSync) getCustomResources(namespaces []string) (map[string][]ReleaseFhr, error) {
+	relInfo := make(map[string][]ReleaseFhr)
+
+	for _, ns := range namespaces {
+		list, err := rs.getNSCustomResources(ns)
+		if err != nil {
+			rs.logger.Log("error", fmt.Errorf("Failure while retrieving FluxHelmReleases in namespace %s: %v", ns, err))
+			return nil, err
+		}
+		rf := []ReleaseFhr{}
+		for _, fhr := range list.Items {
+			relName := release.GetReleaseName(fhr)
+			rf = append(rf, ReleaseFhr{RelName: relName, Fhr: fhr})
+		}
+		if len(rf) > 0 {
+			relInfo[ns] = rf
+		}
+	}
+	return relInfo, nil
+}
+
+// GetCustomResourcesLastDeploy retrieves and stores last event timestamp for FluxHelmRelease resources (FHR)
+// (Tamara: The method is not currently used. I am leaving it here as it will come handy for display on the UI)
+//		output:
+//             map[namespace][FHR name] = int64
+func (rs *ReleaseChangeSync) GetCustomResourcesLastDeploy(namespaces []string) (map[string]map[string]int64, error) {
+	relEventsTime := make(map[string]map[string]int64)
+
+	for _, ns := range namespaces {
+		eventList, err := rs.getNSEvents(ns)
+		if err != nil {
+			return relEventsTime, err
+		}
+		fhrD := make(map[string]int64)
+		for _, e := range eventList.Items {
+			if e.InvolvedObject.Kind == CustomResourceKind {
+				secs := e.LastTimestamp.Unix()
+				fhrD[e.InvolvedObject.Name] = secs
+			}
+		}
+		relEventsTime[ns] = fhrD
+	}
+	return relEventsTime, nil
+}
+
+func (rs *ReleaseChangeSync) shouldUpgrade(currRel *hapi_release.Release, fhr ifv1.FluxHelmRelease) (bool, error) {
+	currVals := currRel.GetConfig().GetRaw()
+	currChart := currRel.GetChart().String()
+
+	// Get the desired release state
+	opts := chartrelease.InstallOptions{DryRun: true}
+	tempRelName := strings.Join([]string{currRel.GetName(), "temp"}, "-")
+	desRel, err := rs.release.Install(rs.release.Repo.ReleasesSync, tempRelName, fhr, "CREATE", opts)
+	if err != nil {
+		return false, err
+	}
+	desVals := desRel.GetConfig().GetRaw()
+	desChart := desRel.GetChart().String()
+
+	// compare values && Chart
+	if strings.Compare(currVals, desVals) != 0 {
+		rs.logger.Log("error", fmt.Sprintf("Release %s: values have diverged due to manual Chart release", currRel.GetName()))
+		return true, nil
+	}
+	if strings.Compare(currChart, desChart) != 0 {
+		rs.logger.Log("error", fmt.Sprintf("Release %s: Chart has diverged due to manual Chart release", currRel.GetName()))
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// existingReleasesToSync determines which Chart releases need to be deleted/upgraded
+// to bring the cluster to the desired state
+func (rs *ReleaseChangeSync) existingReleasesToSync(
+	currentReleases map[string]map[string]int64,
```

Comment:
Is this `int64` ever examined?

Reply:
> I am leaving the signature of the GetCurrentWithDate() method (release.go) as is for future use.

This approach doesn't make sense to me.

This repo should not serve as a notepad for code you suspect you'll need in future. Delete unused code-paths, or someone else will delete it without warning.

If you'd like to store a code snippet you've written for later reuse, [create a Gist](https://gist.github.com/).

Reply:
I made requested changes, which can be put back when need be.